lib/f_core/os/c_file.cpp

```diff
@@ -0,0 +1,68 @@
+#include "f_core/os/c_file.h"
+
+#include <zephyr/fs/fs.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/http/parser.h>
+
+LOG_MODULE_REGISTER(CFile);
+
+CFile::CFile(const char* path, const int flags) : path(path), flags(flags) {
+    fs_file_t_init(&file);
+
+    initStatus = fs_open(&file, path, flags);
+
+    if (initStatus < 0) {
+        LOG_ERR("Error opening %s. %d", path, initStatus);
+    }
+}
+
+CFile::~CFile() {
+    if (initStatus == 0) {
+        fs_close(&file);
+    }
+}
+
+size_t CFile::GetFileSize() const {
```

Comment:
would it make sense to make this a static function that takes a path bc it's fine related but doesn't actually use the fs_file_t that this class wraps?

Reply:
Out of scope for a file class imo